packages/services/src/perps-data-processor.service.ts

```diff
@@ -214,13 +214,51 @@ export class PerpsDataProcessor {
         `[PerpsDataProcessor] Processing agent ${agentId} for competition ${competitionId}`,
       );
 
-      // 1. Fetch account summary from provider
-      const accountSummary = await provider.getAccountSummary(walletAddress);
+      // 1. Check for existing snapshots to determine initial capital
+      let initialCapital: number | undefined;
+      try {
+        const { first } = await this.competitionRepo.getFirstAndLastSnapshots(
+          competitionId,
+          agentId,
+        );
+
+        if (first) {
+          const capitalValue = Number(first.totalValue);
 
-      // 2. Fetch positions from provider
+          // Validate the conversion - protect against NaN or invalid values
+          if (isFinite(capitalValue) && capitalValue >= 0) {
```

Comment:
just curious: when would this ever be non-finite or `<=0`

Reply:
I suppose it might be overly defensive, but pg's numeric type can store special values like NaN which would make isFinite() return false.

As for <0, in theory it should never happen, but since we rely so heavily on these external data providers for our reporting (and as we saw last week with Symphony's spotty API), doesn't hurt to be a bit more defensive